src/rust/uid-allocator/tests/uid_allocator.rs

```diff
@@ -1,31 +1,62 @@
 #![cfg(all(test, feature = "integration_tests"))]
 
 use clap::Parser;
-use rust_proto::graplinc::grapl::api::uid_allocator::v1beta1::{
-    client::UidAllocatorServiceClient,
-    messages::CreateTenantKeyspaceRequest,
+use rust_proto::{
+    client_factory::services::UidAllocatorClientConfig,
+    graplinc::grapl::api::uid_allocator::v1beta1::messages::CreateTenantKeyspaceRequest,
 };
 use uid_allocator::client::CachingUidAllocatorServiceClient;
 
 #[tokio::test]
 async fn test_uid_allocator() -> Result<(), Box<dyn std::error::Error>> {
-    let client_config = uid_allocator::config::UidAllocatorClientConfig::parse();
+    let _guard = grapl_tracing::setup_tracing("uid-allocator integ test")?;
 
     let tenant_id = uuid::Uuid::new_v4();
-
-    let endpoint = format!("http://{}", &client_config.uid_allocator_connect_address);
-    let mut allocator_client = CachingUidAllocatorServiceClient::new(
-        UidAllocatorServiceClient::connect(endpoint).await?,
-        100,
+    tracing::info!(
+        tenant_id = ?tenant_id,
     );
 
+    let client_config = UidAllocatorClientConfig::parse();
+    let mut allocator_client =
+        CachingUidAllocatorServiceClient::from_client_config(client_config, 100).await?;
+
+    tracing::info!("creating keyspace");
     allocator_client
         .create_tenant_keyspace(CreateTenantKeyspaceRequest { tenant_id })
         .await?;
 
-    for i in 1u64..10_000 {
+    // If there were 1 single `uid-allocator` instance we're talking to we'd
+    // expect this to go monotonically increasing upwards (1 to 10,000),
+    // but since we currently have two `uid-allocator` - each reserving 10_000
+    // locally on the server - we can make no promises about the order in which
+    // uids are allocated.
+
+    // Example behavior:
+    // client:  "give me 100 ids for tenant ABC"
+    // server1: "ok i've reserved 1-10_000, here's 1-100"
+    // client:  "give me 100 ids for tenant ABC"
+    // server2: "ok i've reserved 10_001-20_000, here's 10_001-10_101"
+    // client:  "give me 100 ids for tenant ABC"
+    // server1: "I already have 1-10_000 reserved, so heres 101-200"
+
+    tracing::info!("allocating ids");
+    let num_requested_uids: u64 = 11_000;
+    // Just a sanity check that we're not allocating insane uids
```

Comment:
I'd just note that "if you're seeing this test fail, check that the number of instances is still 2 and the preallocation size is still 10k"